aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -357,13 +464,22 @@ bool _is_same_size_as_sparse_csr(
 const SparseCsrTensor& resize_as_sparse_csr_(
     const SparseCsrTensor& self,
     const SparseCsrTensor& src) {
-  TORCH_CHECK(
-      src.is_sparse_csr() && self.is_sparse_csr(),
-      "resize_as_sparse_csr_: layout for self and src must be sparse_csr but got ",
-      self.layout(),
-      " for self, and ",
-      src.layout(),
-      " for src");
+  switch(self.layout()) {
```

Comment:
If we had an `AT_DISPATCH_LAYOUT_SPARSE_COMPRESSED`, akin to the functions in [Dispatch.h](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Dispatch.h), the entirety of
```
switch(self.layout()) {
  case kSparseCsr:
  case kSparseCsc:
  case kSparseBsr:
  case kSparseBsc:
    TORCH_CHECK(
                src.layout() == self.layout(),
                "resize_as_sparse_csr_: layout for self and src must be match but got ",
                self.layout(),
                " for self, and ",
                src.layout(),
                " for src");
    break;
  default:
    TORCH_CHECK(false, "resize_as_sparse_csr_: layout ", self.layout(), " is not supported");
  }
```

 could be written as

```
AT_DISPATCH_LAYOUT_SPARSE_COMPRESSED(self.layout(), "resize_as_sparse_csr_", [&]() {
    TORCH_CHECK(
        src.layout() == self.layout(),
        "resize_as_sparse_csr_: layout for self and src must be match but got ",
         self.layout(),
         " for self, and ",
         src.layout(),
         " for src");
});
```

I think that'd be a useful macro in general. What do you think?

Reply:
This makes sense: it will save/unify us typing `switch-cases` constructs and the `default` case handling.

Reply:
Resolved via https://github.com/pytorch/pytorch/pull/75865